### ISSUE TITLE
pylint ignoring the disable comment in some cases 

### ISSUE THREAD

**Issue by RoniGurvich** [PyCQA/pylint#4799]
### Bug description

```python
it looks like some disable messages are ignored


a = None
condition = True
if condition:
    # pylint: disable=unsupported-assignment-operation
    a['b'] = 'c'  # not raising error
    # pylint: enable=unsupported-assignment-operation
else:
    pass

if not condition:
    pass
else:
    # pylint: disable=unsupported-assignment-operation
    a['b'] = 'c'  # raising error
    # pylint: enable=unsupported-assignment-operation
```

### Command used

```shell
pylint disable_issue.py
```

### Pylint output

```shell
************* Module disable_issue
disable_issue.py:14:4: E1137: 'a' does not support item assignment (unsupported-assignment-operation)

-----------------------------------
Your code has been rated at 3.75/10
```

### Expected behavior

no issues found because it is suppressed

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.6
Python 3.9.6 (default, Jun 29 2021, 05:25:02)
[Clang 12.0.5 (clang-1205.0.22.9)]
```

### OS / Environment

macOS

**Comment by Pierre-Sassoulas** [comment]
Duplicate of #872
